src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -13,8 +13,12 @@
     /// </summary>
     internal class HostingDiagnosticListener : IApplicationInsightDiagnosticListener
     {
+        /// <summary>
+        /// Conversion from stopwatch time (ticks since boot) to local time.
+        /// </summary>
+        private static readonly DateTimeOffset stopwatchEpoch = DateTime.Now - TimeSpan.FromTicks(Stopwatch.GetTimestamp());
```

Comment:
OK, I have switched to using Start/Stop. I had to disable one of the tests until the overloads I added in https://github.com/Microsoft/ApplicationInsights-dotnet/pull/424 are available in a nuget we can consume from here. In testing, I also discovered that we had a precision problem in Stop() (it was rounding durations to the nearest millisecond) which I fixed with https://github.com/Microsoft/ApplicationInsights-dotnet/pull/426.